client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -373,6 +375,17 @@ static String endpoint(String... parts) {
     }
 
     /**
+     * Returns a {@link ContentType} from a given {@link XContentType}.
+     *
+     * @param xContentType the {@link XContentType}
+     * @return the {@link ContentType}
+     */
+    @SuppressForbidden(reason = "Only allowed place to convert a XContentType to a ContentType")
+    static ContentType createContentType(final XContentType xContentType) {
+        return ContentType.create(xContentType.mediaTypeWithoutParameters(), (Charset) null);
```

Comment:
I don't want to force a charset because Elasticsearch exposes many ways to set the source of, let's say, an IndexRequest. It can be a Map converted by Jackson as UTF-8 bytes, but it can also be raw bytes too... And Elasticsearch will rely on Jackson auto-detect encoding to parse this correctly.

I think there are many things to dig here and in the meanwhile setting a `null` charset allows to keep the current behavior.